PROJECT_STRUCTURE.md

```diff
@@ -0,0 +1,86 @@
+![Remixer](https://cdn.rawgit.com/material-foundation/material-remixer/master/docs/assets/lockup_remixer_icon_horizontal_dark_small.svg)
+
+# Project structure
+
+Remixer is built as a gradle project with several submodules. If you are interested in contributing you should know the roles of each submodule to be able to put things in the right place.
+
+## Core and Annotations
+
+Remixer has a core library, `remixer_core`, where the main logic for Remixer is, here is where you can find
+- DataTypes supported by Remixer `com.google.android.libraries.remixer.DataType`
+- Serialization logic `package com.google.android.libraries.remixer.serialization`
+- All variable logic `com.google.android.libraries.remixer.Variable`, `com.google.android.libraries.remixer.ItemListVariable`, `com.google.android.libraries.remixer.RangeVariable`
+- The `com.google.android.libraries.remixer.Remixer` class which aggregates all variables and `package com.google.android.libraries.remixer.sync` which takes care of keeping values in sync across different contexts.
+  - `com.google.android.libraries.remixer.sync.LocalValueSyncing` is a non-persistent version of a `com.google.android.libraries.remixer.sync.SynchronizationMechanism`. Persistent versions are available in the `remixer_storage` submodule, those depend on android-specific classes which are unavailable on this project, because of the reasons explained in the next section.
+- Declaration of all annotations exposed by Remixer `package com.google.android.libraries.remixer.annotation` and the annotation binding classes.
+
+The annotation-processing library, `remixer_annotation`, contains code that should only be run by the Java Compiler (and should never be in the classpath for an APK), this code generates Remixer variables from annotations in `com.google.android.libraries.remixer.annotation`.
+
+
+### Constraints on dependencies
+
+**TL;DR: If you contribute to Remixer, you need to know you cannot depend on any android classes in the `remixer_core` submodule**
+
+Remixer uses annotations heavily, and annotation processing on Android is very difficult. The Android SDK does not include classes required for annotation processing, therefore if you depend on the Android SDK, you cannot use annotation processing. For this reason there are two submodules that are Java libraries instead of Android libraries, `remixer_core` and `remixer_annotation`.
+
+This means that the annotation processing code (in `remixer_annotation`) is unable to generate code that uses any Android class (i.e. can't generate `OnClickListener`s, or use `Actvity` in generated code, etc.) [^generatingtestedcode]. Furthermore, in some cases, where Remixer needs a Context (which should naturally be an instance of `android.app.Activity` or at least `android.content.Context`) we take `java.lang.Object` instead. It is generally very well explained in comments that this is the case.
```

Comment:
Is `[^generatingtestedcode]` supposed to be a footnote? It's not formatted at all in the Preview. You may want to stick with `[1]` which does work.